site/en/blog/longer-esw-lifetimes/index.md

```diff
@@ -35,9 +36,12 @@ Unfortunately, this behavior did not apply to extension events. Extension events
 
 ## What's changed
 
-As of Chrome 110, all events reset the lidle timer and the idle timeout will not occur if there are pending events. In other words, assuming there are no unexpected interruptions, extension service workers will now stay alive as long as they are actively processing events.
+As of Chrome 110, all events reset the lidle timer and the idle timeout will not occur if there are pending events. In other words, assuming there are no unexpected interruptions, extension service workers will now stay alive as long as they are actively processing events. Be aware that:
 
-While this change removes the five minute limit on extension service worker lifetimes, you are still strongly encouraged to design your extensions to be event-based and guard against unexpected termination. Not only should you refrain from keeping your extensions alive indefinitely (which we consider an anti-pattern), you should test your extensions to ensure that they're not doing this unintentionally.
+* The service worker will terminate after 30 seconds if there are no events.
+* The service worker will terminate after 5 minutes if a single request (event or API call) takes longer than 5 minutes to process.
+
+You are still strongly encouraged to design your extensions to be event-based and guard against unexpected termination. Not only should you refrain from keeping your extensions alive indefinitely (which we consider an anti-pattern), you should test your extensions to ensure that they're not doing this unintentionally.
 
 In a future release, we may change how specific events are processed to prevent keeping service workers alive indefinitely, but without shutting down long-running work. In the meantime, make sure your extensions yield when possible and prepare for termination by persisting extension states. 
```

Comment:
This line doesn't make sense to me.

> to prevent keeping service workers alive indefinitely

We don't mind this as long as it's legitimately useful work.

> but without shutting down long-running work

This seems directly contradictory to the long-running task timer, which explicitly _will_ shut down long-running work.

If you want to keep this sentence, maybe:

"Future releases of Chromium may prevent artificially extending service worker lifetimes."

That said, I we could even just remove this whole sentence, and just append the second to the previous paragraph:

"... you should test your extensions to ensure that they're not doing this unintentionally and make sure your extensions yield whenever possible..."

Reply:
I took the second approach. The existing language was Simeon's. Not only do I prefer to stay away from remarks about the future, such promises are also contrary to Google's style guide.